### ISSUE TITLE
Updating the LR at every step 

### ISSUE THREAD

**Issue by pclucas14** [facebookresearch/vissl#219]
## 📚 VISSL Documentation

Hi, 

 I am trying to do a linear warmup with a cosine schedule. I followed the instructions [here](https://github.com/facebookresearch/vissl/blob/a5c973bce3cd778a17a0045dd3ee2f56ef247624/vissl/optimizers/param_scheduler/README.md). While I specified `update_interval=step`, it seems the learning rate update only happens every epoch when I look at the TB plots. 
![Screen Shot 2021-03-05 at 10 21 52 AM](https://user-images.githubusercontent.com/20862882/110135630-baf63580-7d9c-11eb-89cf-cdb15366e575.png)

This is most likely me not understanding how to use the scheduling args, so any documentation related to this would help. I looked [here](https://github.com/facebookresearch/vissl/blob/a5c973bce3cd778a17a0045dd3ee2f56ef247624/vissl/optimizers/param_scheduler/README.md) and on the default config but could not find an answer.

Here is my current optimization config 
```
  OPTIMIZER:
      name: adam
      lr: 0.0001
      weight_decay: 0
      num_epochs: 50
      clip_grad_norm: 1
      head_optimizer_params:
        use_different_lr: False
        use_different_wd: False
      param_schedulers:
        lr:
          auto_lr_scaling:
            auto_scale: false
            base_value: 0.3
            base_lr_batch_size: 256
          name: composite
          schedulers:
            - name: linear
              start_value: 1e-5
              end_value: 1e-4
            - name: cosine
              start_value: 1e-4
              end_value: 0.
              # wave_type: half
              # restart_interval_length: 0.5
              wave_type: full
              is_adaptive: True
              restart_interval_length: 0.334
          interval_scaling: [rescaled, fixed]
          update_interval: step
          lengths: [0.05, 0.95]                 # 100ep # how to split between warmup and cos. anneal?
```

(here is what I am trying do to, for reference)
![Screen Shot 2021-03-05 at 10 20 04 AM](https://user-images.githubusercontent.com/20862882/110135553-a6b23880-7d9c-11eb-9739-7d89e5c5fb1c.png)

**Comment by prigoyal** [comment]
Hi @pclucas14 , thank you for reaching out about this. Can you show / check what the `log.txt` file shows for the learning rate? It might be the case that we didn't update the tensorboard frequently enough !

**Comment by pclucas14** [comment]
Hi Priya, 

it's hard to say from the logs. I'm using small learning rates, so the printing precision may not be enough

```
 1353 INFO 2021-03-07 16:07:40,957 log_hooks.py: 207: Rank: 0; [ep: 1] iter: 3175; lr: 4e-05; loss: 2.52264; btime(ms): 470; eta: 23:06:44; peak_mem: 11709M
 1354 INFO 2021-03-07 16:07:51,308 tensorboard_hook.py: 201: Logging metrics. Iteration 3200
 1355 INFO 2021-03-07 16:07:51,310 log_hooks.py: 207: Rank: 0; [ep: 1] iter: 3200; lr: 4e-05; loss: 2.38588; btime(ms): 470; eta: 23:05:14; peak_mem: 11709M
 1356 INFO 2021-03-07 16:08:01,681 tensorboard_hook.py: 201: Logging metrics. Iteration 3225
 1357 INFO 2021-03-07 16:08:01,683 log_hooks.py: 207: Rank: 0; [ep: 1] iter: 3225; lr: 4e-05; loss: 2.65125; btime(ms): 469; eta: 23:03:46; peak_mem: 11709M
 1358 INFO 2021-03-07 16:08:11,961 tensorboard_hook.py: 201: Logging metrics. Iteration 3250
 1359 INFO 2021-03-07 16:08:11,963 log_hooks.py: 207: Rank: 0; [ep: 1] iter: 3250; lr: 5e-05; loss: 2.82069; btime(ms): 469; eta: 23:02:15; peak_mem: 11709M
 1360 INFO 2021-03-07 16:08:22,351 tensorboard_hook.py: 201: Logging metrics. Iteration 3275
 1361 INFO 2021-03-07 16:08:22,353 log_hooks.py: 207: Rank: 0; [ep: 1] iter: 3275; lr: 5e-05; loss: 2.52566; btime(ms): 468; eta: 23:00:51; peak_mem: 11709M
 1362 INFO 2021-03-07 16:08:32,694 tensorboard_hook.py: 201: Logging metrics. Iteration 3300
 1363 INFO 2021-03-07 16:08:32,695 log_hooks.py: 207: Rank: 0; [ep: 1] iter: 3300; lr: 5e-05; loss: 2.61816; btime(ms): 468; eta: 22:59:26; peak_mem: 11709M
 1364 INFO 2021-03-07 16:08:43,021 tensorboard_hook.py: 201: Logging metrics. Iteration 3325
 1365 INFO 2021-03-07 16:08:43,026 log_hooks.py: 207: Rank: 0; [ep: 1] iter: 3325; lr: 5e-05; loss: 2.52307; btime(ms): 467; eta: 22:58:01; peak_mem: 11709M
```

Let me try with bigger learning rates, and I will get back to you

**Comment by pclucas14** [comment]
As an aside, I'm only having this issue with the Adam optimizer. Using SGD I can see the update happening much more frequently, using the same tensorboard settings

**Comment by pclucas14** [comment]
After scaling up the learning rate this issue seems to go away. I'm not exactly sure what was happening (maybe TB doesn't let you log very small values ?), anyhow closing the issue

**Comment by brdav** [comment]
We observed the same issue. It seems to be caused by rounding the logged learning rate to 5 digits after the decimal point:
https://github.com/facebookresearch/vissl/blob/5fb443f63bceacfebbea1adb3803041cdfe0c38d/vissl/hooks/log_hooks.py#L235
Since learning rates are often below that in order of magnitude, this should be fixed to rounding to e.g. 5 significant digits instead.
